### PR TITLE
Agent cuncurrency primitive

### PR DESCRIPTION
**Pull request by susliko**

### CODE REVIEW

**Comment by susliko** [comment]
@Odomontois 
Ready to discuss any ideas

File: build.sbt [hunk 1]
```diff
@@ -123,7 +123,7 @@ lazy val observable = project.settings(
 lazy val concurrent =
   project dependsOn core settings (
     defaultSettings,
-    libraryDependencies ++= Seq(catsEffect, catsTagless),
+    libraryDependencies ++= Seq(catsEffect, catsTagless, fs2Core),
```
Comment (Odomontois) [inline build.sbt]: I don't like this dependency, is it necessary?

Comment (susliko) [reply build.sbt]: We can implement our own concurrent queue, e.g. Michael-Scott Queue. This would allow us to get rid of fs2

Comment (atapin) [reply build.sbt]: @susliko why not scala stdlib's concurrent queue?

Comment (Odomontois) [reply build.sbt]: @atapin Can you be more detailed?

Comment (atapin) [reply build.sbt]: @Odomontois I mean scala collections. I mistaken with word `concurrent`

**Comment by Odomontois** [comment]
Closing this for now. Feel free to reopen

**Pull request closed without merging**